common/core/web/keyboard-processor/src/text/keyboardProcessor.ts

```diff
@@ -714,9 +714,11 @@ namespace com.keyman.text {
     };
 
     setNumericLayer(device: utils.DeviceSpec) {
-      let layout = this.activeKeyboard.layout(device.formFactor);
-      if(layout.getLayer('numeric')) {
-        this.layerId = 'numeric';
+      if (this.activeKeyboard) {
```

Comment:
This check guards against console exceptions

Reply:
This check feels like it belongs higher up -- in caller within Keyman for Android -- because it must be happening when Keyman is not fully initialised. 

Not that it's entirely a bad thing to have a sanity check here but feels like it is side-stepping a more fundamental problem.

Reply:
Should I have keyboard.html do something like this then?
```
    // Tell KMW to switch to "numeric" layer
    function setNumericLayer() {
      var kmw=window['keyman'];
      if (kmw.core.activeKeyboard) {  // <--- add this
        kmw.setNumericLayer();
      }
    }
```

Reply:
Yes. Although I'd test that `kmw` and `kmw.core` are also initialised (`if (kmw && kmw.core && kmw.core.activeKeyboard)`)